src/utils/matrix.cpp

```diff
@@ -64,6 +64,18 @@ void matrix_mult(matrix_t Pmat, matrix_t Qmat, matrix_t Rmat) {
     }
 }
 
+void matrix_add(matrix_t Amat, matrix_t Bmat, matrix_t Cmat) {
+    size_t n = Amat.rows;
+    double* A = Amat.ptr;
+    double* B = Bmat.ptr;
+    double* C = Cmat.ptr;
+    for (size_t i = 0; i < n; ++i) {
+        for (size_t j = 0; j < n; ++j) {
+            C[n * i + j] = A[n * i + j] + B[n * i + j];
```

Comment:
Could this be unrolled like `copy_block`?

Reply:
In general I didn't do any unrolling in this. I see that `copy_block` contains some unrolling but that was just for experimenting, and it has no benefit whatsoever.